rust/blockstore/src/arrow/blockfile.rs

```diff
@@ -416,127 +417,22 @@ impl<'me, K: ArrowReadableKey<'me> + Into<KeyWrapper>, V: ArrowReadableValue<'me
-        key: K,
-    ) -> Result<Vec<(K, V)>, Box<dyn ChromaError>> {
-        // Get all block ids that contain keys > key from sparse index for this prefix.
-        let block_ids = self.root.sparse_index.get_block_ids_range(
-            prefix..=prefix,
-            (
-                std::ops::Bound::Excluded(key.clone()),
-                std::ops::Bound::Unbounded,
-            ),
-        );
-        let mut result: Vec<(K, V)> = vec![];
-        // Read all the blocks individually to get keys > key.
-        for block_id in block_ids {
-            let block_opt = match self.get_block(block_id).await {
-                Ok(Some(block)) => Some(block),
-                Ok(None) => {
-                    return Err(Box::new(ArrowBlockfileError::BlockNotFound));
-                }
-                Err(e) => {
-                    return Err(Box::new(e));
-                }
-            };
-
-            let block = match block_opt {
-                Some(b) => b,
-                None => {
-                    return Err(Box::new(ArrowBlockfileError::BlockNotFound));
-                }
-            };
-            result.extend(block.get_gt(prefix, key.clone()));
-        }
-        Ok(result)
-    }
-
-    /// Returns all arrow records whose key < supplied key.
-    pub(crate) async fn get_lt(
-        &'me self,
-        prefix: &str,
-        key: K,
-    ) -> Result<Vec<(K, V)>, Box<dyn ChromaError>> {
-        // Get all block ids that contain keys < key from sparse index.
-        let block_ids = self
-            .root
-            .sparse_index
-            .get_block_ids_range(prefix..=prefix, ..key.clone());
-        let mut result: Vec<(K, V)> = vec![];
-        // Read all the blocks individually to get keys < key.
-        for block_id in block_ids {
-            let block_opt = match self.get_block(block_id).await {
-                Ok(Some(block)) => Some(block),
-                Ok(None) => {
-                    return Err(Box::new(ArrowBlockfileError::BlockNotFound));
-                }
-                Err(e) => {
-                    return Err(Box::new(e));
-                }
-            };
-
-            let block = match block_opt {
-                Some(b) => b,
-                None => {
-                    return Err(Box::new(ArrowBlockfileError::BlockNotFound));
-                }
-            };
-            result.extend(block.get_lt(prefix, key.clone()));
-        }
-        Ok(result)
-    }
-
-    /// Returns all arrow records whose key >= supplied key.
-    pub(crate) async fn get_gte(
-        &'me self,
-        prefix: &str,
-        key: K,
-    ) -> Result<Vec<(K, V)>, Box<dyn ChromaError>> {
-        // Get all block ids that contain keys >= key from sparse index.
+        prefix_range: PrefixRange,
+        key_range: KeyRange,
+    ) -> Result<Vec<(K, V)>, Box<dyn ChromaError>>
+    where
+        PrefixRange: RangeBounds<&'prefix str> + Clone,
+        KeyRange: RangeBounds<K> + Clone,
```

Comment:
I think it should be possible to remove the Clone bound by allowing borrowed ranges but I'm not entirely sure how. Methods like `block.get_range()` should accept both borrowed and owned ranges for ease of use. (E.x. I don't want to require users to do `block.get_range(&(0..10))`)